src/omnisharp/logger.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+export class Logger {
+    private _writer: (message: string) => void;
+    private _prefix: string;
+
+    private _indent: string = '    ';
+    private _indentLevel: number = 0;
+    private _atLineStart: boolean = false;
+
+    constructor(writer: (message: string) => void, prefix?: string) {
+        this._writer = writer;
+        this._prefix = prefix;
+    }
+
+    private _appendCore(message: string): void {
+        if (this._atLineStart) {
+            if (this._indentLevel > 0) {
```

Comment:
To be clear, this was supposed to just be a joke based on the fact that the removal of leftpad from NPM broke the internet.